### PR TITLE
DM-23324: Update AT ops procedure (tstn-004)

### PR DESCRIPTION
**Pull request by tribeiro**
Update documentation with the latest information.
Add information about LOVE interface.
Update Jupyter interface image.
Add new version of ATTCS diagram with LATISS.
Add new version of ATTCS diagram with LATISS.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -45,108 +45,15 @@ Network architecture and connectivity
...
-connected to the LSST network (e.g. LSST-WAP) you will have access to the summit network. The
-control network on the other side is only accessible from bastion computers on the summit. These
-bastions are connected to both the campus network and the control network, thus giving users
-access to the control network through :ref:`ssh tunneling <ssh-rules>`.
-
-  .. attention::
-     An important aspect of the control network is that it **does not** have access to the internet.
-     This does create some issues, for instance, to update software on computers connected solely to
-     the control network on the fly.
+and the control network. In principle, if you are at the summit, the base or in Tucson, if you are
```
Comment (patrickingraham) [inline index.rst:48]: This isn't strictly true. Even if I sit in my office in Tucson I have to VPN into the tucson VPN in order to get EFD access.

File: index.rst [hunk 2]
```diff
@@ -45,108 +45,15 @@ Network architecture and connectivity
...
+connected to the LSST network (e.g. LSST-WAP) you will have access to the summit network and the
+control network. IT managed to overlay both networks so that there is no need to use bastion
+hosts to access systems on the control network from LSST network. This considerably simplify
+user access to resources avoiding the need for ssh tunnels.
 
 A list of the host computers IP address can be found
```
Comment (patrickingraham) [inline index.rst]: addresses

File: index.rst [hunk 3]
```diff
@@ -206,36 +116,62 @@ The current system deployment uses individual Jupyter Lab servers for each user
...
 see and use work that was done.
 
+While each user has their only :file:`develop` space, the containers have a shared mount space
+visible to all users, :file:`shared`. Filed placed or edited here by a user in their jupyter server
+will be available/modified to all the other users.
+
 The :file:`repos` directory, on the other hand, contains some basic repos that ships with the
-container with the T&S software used to power the control system. Any data in this directory,
+notebook server containing the T&S software used to power the control system. Any data in
+this directory,
```
Comment (patrickingraham) [inline index.rst:174]: Maybe emphasize this (italics or something)

File: index.rst [hunk 4]
```diff
@@ -206,36 +116,62 @@ The current system deployment uses individual Jupyter Lab servers for each user
...
 or in the home folder, will be lost if the container is restarted. It is advisable to only keep
 important data inside the user designated folder (e.g. :file:`develop`).
 
+It is also possible to access data taken with the LATISS instrument in the notebook server.
+The data is immediately available in a butler instance in
+:file:`/mnt/dmcs/oods_butler_repo/repo/`. This mount point is read-only by all users but
+inside that there are a couple of shared mount places for users to save calibration
+and reduced data (:file:`/mnt/dmcs/oods_butler_repo/repo/CALIB` and
+:file:`/mnt/dmcs/oods_butler_repo/repo/rerun`).
+
```
Comment (patrickingraham) [inline index.rst:184]: Do we have example notebooks we can link here?

File: index.rst [hunk 5]
```diff
@@ -206,36 +116,62 @@ The current system deployment uses individual Jupyter Lab servers for each user
...
 .. _love:
 
 LSST Operations and Visualization Environment (LOVE)
 ----------------------------------------------------
 
-.. note::
-    TBD
+The LOVE interface is available at the summit on the following address;
```
Comment (patrickingraham) [inline index.rst:190]: Maybe note that this is still in active development and is not yet stable?

File: index.rst [hunk 6]
```diff
@@ -258,27 +194,24 @@ responsible for capturing all SAL traffic, serialize it in avro an send it over
...
    :name: fig-attcs
-   :target: ../_images/ATTCS.jpg
+   :target: ../_images/ATTCS-2.jpg
    :alt: AuxTel components
 
 
+These components are grouped into high-level components that, although independent,
+work together logically. In the case of AT, these are the Auxiliary Telescope -
+Telescope Control System (ATTCS) and the LSST Auxiliary Telescope Image and
+Slit less Spectrograph (LATISS).
```
Comment (patrickingraham) [inline index.rst:241]: you mean the LATISS control system no? or the class?
Maybe reword to say that the telescope and instrument are controlled via two different systems controlled using two different classes.

File: index.rst [hunk 7]
```diff
@@ -291,105 +224,100 @@ Startup procedure
 
 At the end of the day, before observations starts, most CSCs will be unconfigured and
```
Comment (patrickingraham) [inline index.rst]: telescope related CSCs? All CSCs?

File: index.rst [hunk 8]
```diff
@@ -291,105 +224,100 @@ Startup procedure
...
 in ``STANDBY`` state. The first step in starting up the system is to enable all CSCs.
-Putting a CSC in the ``ENABLED`` state required the transition from ``STANDBY`` to
+Putting a CSC in the ``ENABLED`` state requires the transition from ``STANDBY`` to
 ``DISABLED`` and then from ``DISABLED`` to ``ENABLED``. When transitioning from
 ``STANDBY`` to ``DISABLED`` it is possible to provide a ``settingsToApply`` that selects
 a configuration for the CSC. Some CSCs won't need any settings while others will.
 It is possible to check what are the available settings by looking at the ``settingVersions``
-event.
+event. Alternatively, it is also possible to let the high level control scripts to
```
Comment (patrickingraham) [inline index.rst]: maybe mention that the settingsVersions is easily viewed from the EFD? most operators won't be writing code to grab that event?

File: index.rst [hunk 9]
```diff
@@ -291,105 +224,100 @@ Startup procedure
...
-    from lsst.ts import salobj
+    domain = salobj.Domain()
+    attcs = ATTCS(domain)
+    latiss = LATISS(domain)
 
-Setup remotes for all the AT components
+    await asyncio.gather(attcs.start_task, latiss.start_task)
 
-::
+    await attcs.enable()
```
Comment (patrickingraham) [inline index.rst:305]: There is also a series of documentation for these classes that exist only in the class no? One's that people can read using `help(attcs)` ?

File: index.rst [hunk 10]
```diff
@@ -424,7 +352,7 @@ Initializing ``ATTCS`` class.
 
 Run the slew task. This task will only finish when the telescope and the dome are
```
Comment (patrickingraham) [inline index.rst]: To slew the telescope and dome to a target, you should use the slew_icrs task.

File: index.rst [hunk 11]
```diff
@@ -446,148 +374,84 @@ requested time, which will change as the telescope track the object.
 
     await attcs.slew_icrs(ra="20:25:38.85705", dec="-56:44:06.3230", rot_pos=0., target_name="Alf Pav")
 
-As with the :ref:`startup` procedure, this task has not been tested at the telescope yet.
-For now the sanctioned procedure is to execute the slew and track by commanding the pointing
-component individually. This also means the user have to handle the rotator angle
-computations. In this mode we only support setting the rotator position to a certain angle.
-Due to some binding issues we have been trying to keep the rotator as close to zero as possible.
+Note that this task still experience some undesired behaviour, like hanging in until it times out
+or returning before the end of a slew. These issues are mainly due to issues with the `ATMCS`
```
Comment (patrickingraham) [inline index.rst]: Aren't these fixed? And do we have tickets/info that we've given to Rolando to fix it??

Comment (patrickingraham) [reply index.rst]: Feel free to create a ticket and put me as the assignee...

File: index.rst [hunk 12]
```diff
@@ -446,148 +374,84 @@ requested time, which will change as the telescope track the object.
...
+either because it still fails to comply with our system architecture (like sending events in
+specific order) or due to issues with the internal control algorithms.
```
Comment (patrickingraham) [inline index.rst]: If you mention there is an issue then you have to talk about what to do when it fails...

File: index.rst [hunk 13]
```diff
@@ -446,148 +374,84 @@ requested time, which will change as the telescope track the object.
...
-    await asyncio.sleep(1.)
-    await atptg.cmd_pointAddData.start()
-    await asyncio.sleep(1.)
-    await atptg.cmd_pointCloseFile.start()
+    data_ref = butler.dataRef('raw', **dict(visit=object_data_id_list[0]))
 
+    exposure = isrTask.runDataRef(data_ref).exposure
 
 
 .. _liveview:
```
Comment (patrickingraham) [inline index.rst:521]: You have to mention somewhere here how to change the M3 position :-) and which port the instrument is on.

File: index.rst [hunk 14]
```diff
@@ -802,77 +682,105 @@ Again, you will need to replace ``<Component>`` by the name of the CSC.
...
+    might work) connect to 'atmcs-dev.cp.lsst.org' on port 5900
 #. Enter credentials (ask Patrick or Tiago)
-#. If the GUI is not already open, then open internet explorer and enter the following address in the address bar.
+#. If the GUI is not already open, then open internet explorer and enter the
+    following address in the address bar.
    
    .. code-block:: python 
         
-	http://192.168.1.47:8000/atmcs.html
+	http://atmcs-crio.cp.lsst.org:8000/atmcs.html
```
Comment (patrickingraham) [inline index.rst:808]: Maybe add that the low level GUIs for the spectrograph and dome are also there?

File: index.rst [hunk 15]
```diff
@@ -802,77 +682,105 @@ Again, you will need to replace ``<Component>`` by the name of the CSC.
...
+(we think) the hexapod TCP/IP connection goes down. To reset it, one must
+connect a serial port to the device, establish a connection using the (windows)
+PIMikroMove software, close the connection, then power cycle the controller.
+Power cycling can be done remotely (:ref:`using the switched PDU
+<telescope_cabinet_pdu>`). Until this problem is resolved, we've left a
+permanent serial (RS-232) connection to a local windows machine.
 
 Follow these steps to re-establish TCP/IP connection:
 
 #. Establish VNC connection :ref:`which is the same as the ATMCS GUI VNC shown here <atmcs_gui>`.
```
Comment (patrickingraham) [inline index.rst:836]: This is now wrong, you have to open a teamviewer to Mike Warner's PC.

File: index.rst [hunk 16]
```diff
@@ -802,77 +682,105 @@ Again, you will need to replace ``<Component>`` by the name of the CSC.
...
 #. Open PIMikroMove software from start menu
```
Comment (patrickingraham) [inline index.rst:837]: This is broken too, have to use the PITerminal software

File: index.rst [hunk 17]
```diff
@@ -890,10 +800,15 @@ More details to follow.
...
-In the event that a controller in the cabinet needs power cycling remotely, this may be done by logging into the switchable PDU mounted in the cabinet. The IP and connection info can be found `here <https://confluence.lsstcorp.org/x/qw6SBg>`_
+In the event that a controller in the cabinet needs power cycling remotely,
+this may be done by logging into the switchable PDU mounted in the cabinet.
+The IP and connection info can be found `here <https://confluence.lsstcorp.org/x/qw6SBg>`_
 
-* Channel 1 is connected to the main 24V supply. This will power off the cRIO (and possibly the Copley controllers, Pilz Device, and Smart Relay).
-* Channel 2 is connected to powerbar in bottom of cabinet, which has the 220V connection to the mount (which powers the Embedded PC for the Collimation Camera) as well as the hexapod connected to it.
+* Channel 1 is connected to the main 24V supply. This will power off the cRIO
+(and possibly the Copley controllers, Pilz Device, and Smart Relay).
+* Channel 2 is connected to powerbar in bottom of cabinet, which has the 220V
```
Comment (patrickingraham) [inline index.rst]: This is where the ion pump is now connected and everything on the spectrograph. Put a huge warning here saying not to touch it.

File: index.rst [hunk 18]
```diff
@@ -902,15 +817,28 @@ In the event that a controller in the cabinet needs power cycling remotely, this
...
+rotating part of the enclosure loses connection with the bottom box and may be
+blocking the connection). If the CSC is connected and in disabled or enabled
+state, then this will be shown in the `scbLink` event (must verify). Also, t
+his can be seen in the Main Box Dome Control LabVIEW Remote on the ATMCS
+machine as the `TopComms` light in the bottom left corner.
+
+  * Press the reset button on the cRIO inside the electrical cabinet on the
+    rotating part of the dome (near the lower shutter) to resolve this issue
+
+* The Main cRIO (located in the dome electrical cabinet on the first floor) is
```
Comment (patrickingraham) [inline index.rst]: This has been fixed.

**Review by patrickingraham — CHANGES_REQUESTED** [review]

**Comment by patrickingraham** [comment]
Forgot to mention that when building the document I got the following error.
image

File: index.rst (hunk 8 above)
Comment (tribeiro) [reply index.rst]: I mentioned the EFD+Chronograf but operators will probably never get to see this. The user/operation only need to know this if they want to override the `recommended` settings, as the high level class selects it.

File: index.rst (hunk 9 above)
Comment (tribeiro) [reply index.rst:305]: yes

File: index.rst (hunk 15 above)
Comment (tribeiro) [reply index.rst:836]: Can you update this in a different ticket? not sure I know what to do now... I'll make a note that this needs updating.

File: index.rst (hunk 18 above)
Comment (tribeiro) [reply index.rst]: has it really? I think we noticed that it has not....

File: index.rst [hunk 19]
```diff
@@ -202,40 +112,101 @@ We envision that Jupyter Lab servers will be a fundamental part of LSST control
...
-(:cite:`LDM-542`) environment to manage user servers and environment.
-
-The first step to access a Jupyter Lab server is to tunnel using the IP and token information
-provided for each user by the Telescope & Site team point of contact. Once the ssh tunnel is
-up it should be possible to open a web page on ``localhost:<local-port>``. After entering
-the provided token, you should see the :ref:`Jupyter Lab interface <fig-jupyter-interface>`.
+.. important::
+    As of 2020/01 the old user-based Jupyter Lab server is retired.
+
+For that, users have access to a full-featured `LSST Science Platform instance at the summit`_
```
Comment (patrickingraham) [inline index.rst:118]: we don't have the full features.... missing all sorts of catalogues and stuff. Maybe call it a "streamlined instance"

File: index.rst (hunk 15 above)
Comment (patrickingraham) [reply index.rst:836]: Sure

File: index.rst (hunk 16 above)
Comment (patrickingraham) [reply index.rst:837]: will also fix this in another ticket

File: index.rst [hunk 20]
```diff
@@ -890,27 +860,49 @@ More details to follow.
 Telescope Cabinet Switchable PDU
 --------------------------------
 
-In the event that a controller in the cabinet needs power cycling remotely, this may be done by logging into the switchable PDU mounted in the cabinet. The IP and connection info can be found `here <https://confluence.lsstcorp.org/x/qw6SBg>`_
+.. warning::
+    DO NOT TRY THIS.
+    This PDU server is where the camera ion pump and other critical components
```
Comment (patrickingraham) [inline index.rst:865]: This is no longer true :-) But i can fix this in another ticket when i make the other updates.

**Review by patrickingraham — APPROVED** [review]
very minor comments. please look at them then merge when ready
